packages/common/src/audius-query/types.ts

```diff
@@ -147,13 +156,50 @@ export type QueryHookOptions = {
   debounce?: number
 }
 
+export type PaginatedQueryHookOptions = QueryHookOptions & {
+  /**
+   * Page size to use.
+   * NOTE: if you change this value after the hook was initialized you will cause a cache miss and it will re-fetch data.
+   */
+  pageSize: number
+  /**
+   * Toggles single page data mode. If true the hook will only return the current page's data. Equivalent to usePaginatedQuery in the past
+   */
+  singlePageData?: boolean //
+}
+
 export type QueryHookResults<Data> = {
   data: Data
   status: Status
   errorMessage?: string
   forceRefresh: () => void
 }
 
+export type PaginatedQueryHookResults<Data extends []> =
+  QueryHookResults<Data> & {
+    /**
+     * Tracks load state for new pages loading in (not the initial page; initial page can be tracked via status).
+     * Only used after calling loadMore() first
+     */
+    isLoadingMore: boolean
+    /**
+     * Returns false if we get data back that is less than the page size
```

Comment:
that's an implementation detail. I'd say
> True if the endpoint may have more data available. Indicates that more pages can be fetched

Reply:
True, I guess I mainly wanted to point out the detail since it was helpful to understand that there is a edge case where it says `hasMore=true` but you can still get no data back (aka current page is the end and is a perfect page interval).
I think I'll just add that as an added note and leave off the detail